src/main/java/org/acm/auth/managers/ReactionManager.java

```diff
@@ -0,0 +1,40 @@
+package org.acm.auth.managers;
+
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.channel.priv.PrivateChannelCreateEvent;
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent;
+import net.dv8tion.jda.api.events.message.react.MessageReactionRemoveEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.acm.auth.utils.ReactionRoles;
+import org.jetbrains.annotations.NotNull;
+
+public class ReactionManager extends ListenerAdapter {
+
+    @Override
+    public void onGuildMessageReactionAdd(@NotNull GuildMessageReactionAddEvent event) {
+        if (event.getUser().isBot())
+            return;
+
+        Role role = ReactionRoles.getReactionRole(event.getMessageId(), event.getReactionEmote().getEmoji());
```

Comment:
What happens when there's no reaction role set for a message? This would probably be **null**. :cry:

Reply:
😢